rct229/ruleset_functions/baseline_systems/baseline_hvac_sub_functions/is_hvac_sys_preheat_fluid_loop_attached_to_boiler.py

```diff
@@ -0,0 +1,56 @@
+from rct229.data.schema_enums import schema_enums
+from rct229.utils.assertions import getattr_
+from rct229.utils.jsonpath_utils import find_all, find_one_with_field_value
+
+FLUID_LOOP = schema_enums["FluidLoopOptions"]
+
+
+def is_hvac_sys_preheat_fluid_loop_attached_to_boiler(rmi_b, hvac_b_id):
+    """Returns True if the fluid loop associated with preheat system associated with the HVAC system is attached to a boiler.
+    Returns False if this is not the case.
+
+    Parameters
+    ----------
+    rmi_b : json
+        RMD at RuleSetModelInstance level
+    hvac_b_id : str
+        The HVAC system ID.
+
+    Returns
+    -------
+    bool
+        True: preheat system is attached to a boiler
+        False
+    """
+    is_hvac_sys_preheat_fluid_loop_attached_to_boiler_flag = False
+    boilers = find_all("$.boilers[*]", rmi_b)
+    loop_boiler_dict = dict()
+    for boiler_b in boilers:
+        loop_id = getattr_(boiler_b, "boiler", "loop")
+        if not loop_id in loop_boiler_dict.keys():
+            loop_boiler_dict[loop_id] = []
+        loop_boiler_dict[loop_id].append(boiler_b)
+
+    # Get the hvac system
+    hvac_b = find_one_with_field_value(
```

Comment:
I wonder if `find_exactly_one_with_field_value` should be used here.  The RDS seems to assume that there exists an hvac system with id =  hvac_b_id.  The logic here would return False if the havac system does not exist, but I'm thinking that `find_exactly_one_with_field_value` raising an exception might be more appropriate.